app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,104 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task_id = params[:id]
+    @task = Task.find_by(id: @task_id)
+
+    if @task.nil?
+      redirect_to root_path
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to task_path(@task)
+      return
+    else
+      render :new
+      return
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if@task.nil?
+      redirect_to root_path
+      return
+    end
+  end 
+
+  def update
+    @task = Task.find_by(id: params[:id])
+
+    if@task.nil?
+      redirect_to root_path
+      return
+    end
+
+    @task = Task.find_by(id: params[:id])
+    if @task.update(task_params)
+      redirect_to task_path(@task)
+      return
+    else
+      render :edit
+      return
+    end
+  end
+
+  def destroy
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to root_path
+      return
+    end
+
+    @task.destroy
+
+    redirect_to root_path
+    return
+  end
+
+  def completed
```

Comment:
Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent. Also, it is a good practice to name custom actions with a verb.